src/uproot/source/fsspec.py

```diff
@@ -132,10 +143,13 @@ def chunks(
         self._num_requested_bytes += sum(stop - start for start, stop in ranges)
 
         chunks = []
+        # _cat_file is async while cat_file is not
+        use_async = (
+            self._fs.async_impl and type(self._executor).__name__ == "LoopExecutor"
+        )
+        cat_file = self._fs._cat_file if use_async else self._fs.cat_file
         for start, stop in ranges:
-            future = self._executor.submit(
-                self._fs.cat_file, self._file_path, start, stop
-            )
+            future = self._executor.submit(cat_file, self._file_path, start, stop)
             chunk = uproot.source.chunk.Chunk(self, start, stop, future)
             future.add_done_callback(uproot.source.chunk.notifier(chunk, notifications))
             chunks.append(chunk)
```

Comment:
Submitting all of the requests on what may be the event loop, may be the `ThreadPoolExecutor`, and may be a `TrivialExecutor`, wrapped up in something that has the interface of an executor.

The extra indirection might not be necessary. If we're going to switch between three (or two?) possible methods, why not do it right here?

Again, that's just something to think about.